common/src/test/java/org/jboss/test/metadata/javaee/AbstractJavaEEEverythingTest.java

```diff
@@ -944,6 +959,207 @@ protected void assertMailSessions(String prefix, MailSessionsMetaData metaDatas,
         }
     }
 
+    protected void assertContextServices(String prefix, ContextServicesMetaData metaDatas, Mode mode, boolean full, JavaEEVersion version) {
+        if (version != JavaEEVersion.V10 || !full) {
+            assertNull(metaDatas);
+            return;
+        }
+        assertNotNull(metaDatas);
+        assertTrue(metaDatas.size() < 4);
+        prefix = prefix + "ContextService";
+        int count = 1;
+        for (ContextServiceMetaData metaData : metaDatas) {
+
+            assertNotNull(metaData);
+            final String metaDataPrefix = prefix + count;
+
+            final Descriptions desc = metaData.getDescriptions();
+            final String name = metaData.getName();
+            final Set<String> cleared = metaData.getCleared();
+            final Set<String> propagated = metaData.getPropagated();
+            final Set<String> unchanged = metaData.getUnchanged();
+            final PropertiesMetaData properties = metaData.getProperties();
+
+            assertEquals(metaDataPrefix + "-name", name);
+            if (count<3) {
+                // desc
+                assertNotNull(desc);
+                Description[] descArr = desc.value();
+                assertNotNull(descArr);
+                assertEquals(1, descArr.length);
+                assertEquals(metaDataPrefix + "-desc", descArr[0].value());
+                // cleared
+                assertNotNull(cleared);
+                assertEquals(cleared.size(), 1);
+                assertEquals(metaDataPrefix + "-cleared", cleared.iterator().next());
+                // propagated
+                assertNotNull(propagated);
+                assertEquals(propagated.size(), 1);
+                assertEquals(metaDataPrefix + "-propagated", propagated.iterator().next());
+                // unchanged
+                assertNotNull(unchanged);
+                assertEquals(unchanged.size(), 1);
+                assertEquals(metaDataPrefix + "-unchanged", unchanged.iterator().next());
+                // properties
+                assertProperties(metaDataPrefix,2,properties);
+            } else {
```

Comment:
@emmartins test XML files only include two examples of configurations, however, the test is asserting expecting at least three or more. 

Do you want to include empty XML element markup to test the defaults or are you fine as it is? I'm not sure if this was intentional, looking at the code the intention seems to test the defaults.

Reply:
copy/paste and can't find why others are like that too, i.e. same test logic and test xml, please let me know if you want me to change this

Reply:
Not a mandatory step to me, but doing it will ensure we are also testing the defaults (there are not too many) and we will not leave unused code.